### PR TITLE
Feature/oe 121

### PR DESCRIPTION
**Pull request by Ron-Junior**
## Goal

Validate if exist the same data

## Tests

- [ ] go to one-energy
- [ ] login with admin or department manager account
- [ ]  go to the admin section and open manage compensation
- [ ]  try add two records with same ROLE + SYSTEMS INSTALLED + DEPARTMENT
- [ ]  it should show a warning that already exists

## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Http/Controllers/Castle/RatesController.php [hunk 1]
```diff
@@ -6,10 +6,12 @@
 use App\Models\Department;
 use App\Models\Rates;
 use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
 use Illuminate\Http\Request;
 
 class RatesController extends Controller
 {
+    use RefreshDatabase;
```
Comment (iammayron) [inline app/Http/Controllers/Castle/RatesController.php:14]: I think that you made a mistake putting this class here. That class is a testing class and must not be used in controllers.

File: resources/views/castle/rate/create.blade.php [hunk 2]
```diff
@@ -34,7 +34,7 @@
                         </div>
                     @endif
                     <div class="md:col-span-3 col-span-2">
-                        <x-input label="Systems Installed" name="time"></x-input>
+                        <x-input type="number" label="Systems Installed" name="time"></x-input>
```
Comment (iammayron) [inline resources/views/castle/rate/create.blade.php:37]: You don't need the </x-input> closing tag, you can simply close the tag typing a / before closing the opening tag.

File: app/Models/Rates.php [hunk 3]
```diff
@@ -23,6 +23,11 @@ public function department()
         return $this->belongsTo(Department::class);
     }
 
+    public function alreadyExists()
+    {
+        return Rates::where('id', '!=', $this->id)->whereRole($this->role)->whereDepartmentId($this->department_id)->where('time', $this->time)->get()->count() > 0;
```
Comment (iammayron) [inline app/Models/Rates.php:28]: You don't need this `> 0` in your code, the count function will return 0 or a greater number, so this result will be validated correctly in your conditional.

File: tests/Feature/Castle/RateTest.php [hunk 4]
```diff
@@ -101,6 +101,31 @@ public function it_should_store_a_new_rate()
...
+              'name'          => 'rate',
+              'time'          => 25,
+              'rate'          => 2.5,
+              'department_id' => $department->id,
+              'role'          => 'Sales Rep',
+          ];
+
+          $this->actingAs($departmentManager);
+
+          $this->post(route('castle.rates.store'), $data);
```
Comment (iammayron) [inline tests/Feature/Castle/RateTest.php:122]: You must record the first Rate using a factory and the second one must use the post action, that is the correct way.

**Review by iammayron — CHANGES_REQUESTED** [review]

**Review by iammayron — APPROVED** [review]
